src/mscorlib/shared/System/Memory.cs

```diff
@@ -315,6 +315,16 @@ public unsafe MemoryHandle Retain(bool pin = false)
         /// </summary>
         public bool TryGetArray(out ArraySegment<T> arraySegment)
         {
+            if (_length == 0)
+            {
+#if FEATURE_PORTABLE_SPAN
+                arraySegment = new ArraySegment<T>(Array<T>.Empty);
```

Comment:
I do not think `Array<T>.Empty` is portable. It should be `SpanHelpers.PerTypeValues<T>.EmptyArray`